controllers/resource/upserter_test.go

```diff
@@ -415,7 +418,45 @@ func Test_upserter_UpsertDoguPVCs(t *testing.T) {
 		// then
 		require.ErrorIs(t, err, assert.AnError)
 	})
-	t.Run("fail when creating a dogu pvc", func(t *testing.T) {
+
+	t.Run("fail when pvc already exists and retrier timeouts", func(t *testing.T) {
+		// given
+		doguResource := readLdapDoguResource(t)
+		dogu := readLdapDogu(t)
+		dogu.Volumes = nil
+
+		client := external.NewClient(t)
+		key := doguResource.GetObjectKey()
+		key.Name = doguResource.GetReservedPVCName()
+		client.On("Get", mock.Anything, key, &v1.PersistentVolumeClaim{}).Return(nil).Run(func(args mock.Arguments) {
+			pvc := args.Get(2).(*v1.PersistentVolumeClaim)
+			now := metav1.Now()
+			pvc.SetDeletionTimestamp(&now)
+		})
+
+		generator := mocks.NewDoguResourceGenerator(t)
+		generator.On("CreateReservedPVC", doguResource).Return(readLdapDoguExpectedReservedPVC(t), nil)
+		upserter := upserter{
+			client:    client,
+			generator: generator,
+		}
+		oldTries := maximumTriesWaitForExistingPVC
+		maximumTriesWaitForExistingPVC = 2
+		defer func() {
+			maximumTriesWaitForExistingPVC = oldTries
+		}()
+
+		// when
+		_, err := upserter.UpsertDoguPVCs(context.Background(), doguResource, dogu)
+
+		// then
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "failed to wait for existing pvc ldap-reserved to terminate: the maximum number of retries was reached: pvc ldap-reserved still exists")
+	})
+
+	resourceNotFoundError := apierrors.NewNotFound(schema.GroupResource{Group: "", Resource: ""}, "")
+
+	t.Run("fail when generate a dogu pvc", func(t *testing.T) {
```

Comment:
I do not understand this test description. Who failed to generate a PVC and why?